bus-mapping/src/evm/opcodes/calldatacopy.rs

```diff
@@ -98,52 +99,68 @@ fn gen_copy_event(
 ) -> Result<CopyEvent, Error> {
     let rw_counter_start = state.block_ctx.rwc;
 
-    let memory_offset = geth_step.stack.nth_last(0)?;
+    let memory_offset = geth_step.stack.nth_last(0)?.low_u64();
     let data_offset = geth_step.stack.nth_last(1)?;
     let length = geth_step.stack.nth_last(2)?.as_u64();
 
     let call_data_offset = state.call()?.call_data_offset;
     let call_data_length = state.call()?.call_data_length;
 
-    // Get low Uint64 of offset.
-    let dst_addr = memory_offset.low_u64();
     let src_addr_end = call_data_offset.checked_add(call_data_length).unwrap();
-
-    // Reset offset to call_data_length if overflow, and set source start to the
-    // minimum value of offset and call_data_length.
     let src_addr = u64::try_from(data_offset)
         .ok()
-        .and_then(|offset| offset.checked_add(call_data_offset))
+        .and_then(|s| s.checked_add(call_data_offset))
         .unwrap_or(src_addr_end)
         .min(src_addr_end);
+    let dst_addr = memory_offset;
 
     let mut exec_step = state.new_step(geth_step)?;
```

Comment:
This `exec_step` will capture the memory RW operations, but it is discarded at the end of this function. Shouldn’t it be the same `exec_step` from `gen_associated_ops` above?

Reply:
I think you are right,  it is historical writing style,  can see it almost everywhere,

Reply:
To conclude: the operations are still recorded in the state. They are strangely not recorded in the event, but it is not needed in EVM circuit assignments anyway.